### PR TITLE
Getting started with Spring Data MongoDB in Kotlin

### LINKED ISSUE #1387: Getting started with Spring Data MongoDB with Kotlin
### Brief Summary: 
Spring Data MongoDB provides an integration between Spring Boot application and the MongoDB document database. In the article I will explain all the concepts need to create a Spring Boot application that integrates with MongoDB document database. I will be creating a simple Spring Boot application in the article to enable the reader to gain practical knowledge on how to use Spring Data MongoDB in a real application.

### Key Takeaways: 
1. What is Spring Data MongoDB.
2. How to use Spring Data MongoDB.
3. Creating Rest APIs in Kotlin with Spring Data MongoDB.

### References: 
N/A

### PR DESCRIPTION
**Pull request by paulodhiambo**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1387

### CODE REVIEW

**Review by LinusMuema — COMMENTED** [review]
Hi @paulodhiambo 
This is a great article, really clear. Just a few suggestions and the rest will be okay

File: articles/spring-data-mongodb/index.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+    //delete todo
+    @DeleteMapping("/{id}")
+    fun deleteTodo(@PathVariable("id") id: String): ResponseEntity<Any> {
+        return ResponseEntity(todoService.deleteTodo(id), HttpStatus.NO_CONTENT)
+    }
+
+}
+```
+- `@CrossOrigin` annotation is used to configure the allowed origins.
```
Comment (LinusMuema) [inline articles/spring-data-mongodb/index.md:87]: You should also mention that it is the same as cors for the developers not experienced with cross origin issues

File: articles/spring-data-mongodb/index.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+}
+```
+### Configurations
+In the resources directory, open the `application.properties` file in your IDE and add the configurations below.
+```yaml
+spring.data.mongodb.database=todo_db
+spring.data.mongodb.port=27017
+
+```
+- `spring.data.mongodb.database=todo_db` indicates that our database name is `todo_db`.
```
Comment (LinusMuema) [inline articles/spring-data-mongodb/index.md:162]: It would be great if you wrote on how to connect the application to the online mongodb clusters

**Comment by paulodhiambo** [comment]
@LinusMuema I have made the suggested changes.

**Comment by LinusMuema** [comment]
All changes are good @paulodhiambo 

@hectorkambow, you can give this a final review

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**